tests/integration/issues/github_1229/test_sharding_empty_index.py

```diff
@@ -0,0 +1,47 @@
+import os
+
+import numpy as np
+
+from jina import Flow, Document
+
+
+def test_sharding_empty_index(random_workspace_name):
+    print("WORKSPACE =", random_workspace_name)
+    os.environ['JINA_TEST_1229_WORKSPACE'] = os.path.abspath(random_workspace_name)
+    num_shards = 2
+
+    f = Flow() \
+        .add(
+        uses='vectorindexer.yml',
+        shards=num_shards,
+        parallel=2,
+        separated_workspace=True,
+        uses_after='_merge_all'
+    )
+
+    num_docs = 2
+    data = []
+    for i in range(num_docs):
+        doc = Document()
+        doc.content = f'random {i}'
+        doc.embedding = np.array([i])
+        doc.update_id()
+        data.append(doc)
+
+    with f:
+        f.index(data)
```

Comment:
to make more explicit that only one shard will have data, add `f.index(data, batch_size=2)` like this u are sure that the request is not split in 2

Reply:
Just to make sure, have you locally tested this fails without ur changes in current master?

Reply:
Hm good point. It seems like the integration test is passing even without my change. However, the multires lyrics example at this commit (https://github.com/jina-ai/examples/blob/96863bec5de24b589fab943bad92fde5328a3ef0/multires-lyrics-search/tests/test_multireslyrics.py#L18) does fail without it and pass with it. That's the original experiment where we noticed the issue

Reply:
Try with `batch_size=2`

Reply:
I reduced the nr of docs to 1 and it still passed

Reply:
then we need to understand why this is passing in master

Reply:
See my comment below